spec/support/request/shop_workflow.rb

```diff
@@ -29,6 +29,15 @@ def toggle_accordion(name)
   end
 
   def add_variant_to_order_cycle(exchange, variant)
+    oc = exchange.order_cycle
+    supplier = variant.product.supplier
+    # An order cycle needs an incoming exchange for a supplier
+    # before having its products. Otherwise the data will be inconsistent and
+    # and not all needed enterprises are loaded into the shop page.
+    if oc.exchanges.from_enterprise(supplier).incoming.empty?
+      create(:exchange, order_cycle: oc, incoming: true,
+                        sender: supplier, receiver: oc.coordinator)
+    end
```

Comment:
Good thinking. Perhaps extract this new block of code to a private method, something along the lines of `ensure_supplier_exchange`.